sdk/src/main/java/com/horizen/account/transaction/EthereumTransaction.java

```diff
@@ -385,6 +386,7 @@ public synchronized AddressProposition getFrom() {
                         )
                 );
             } catch (Exception e) {
+                // TODO: is this a valid case? what are supposed to do with a TX that has a signature we cannot process?
```

Comment:
this may happen, so we should reject this Tx later as invalid.
That was also a reason why we kept Message.from field as `Optional`

Reply:
I think a transaction is rejected based on this before it is turned into a message, also see the other comment.

Reply:
So basically this case is "Invalid Signature"?

Reply:
yes: signature is present, but for some reason "from" address was not derived from it